examples/api/sygus-fun.cpp

```diff
@@ -0,0 +1,127 @@
+/*********************                                                        */
+/*! \file sygus-fun.cpp
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Abdalrhman Mohamed, Andrew Reynolds
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2019 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief A simple demonstration of the Sygus API.
+ **
+ ** A simple demonstration of how to use the Sygus API to synthesize max and min
+ ** functions. Here is the same problem written in Sygus V2 format:
+ **
+ ** (set-logic LIA)
+ **
+ ** (synth-fun max ((x Int) (y Int)) Int
+ **   ((Start Int) (StartBool Bool))
+ **   ((Start Int (0 1 x y
+ **                (+ Start Start)
+ **                (- Start Start)
+ **                (ite StartBool Start Start)))
+ **    (StartBool Bool ((and StartBool StartBool)
+ **                     (not StartBool)
+ **                     (<= Start Start)))))
+ **
+ ** (synth-fun min ((x Int) (y Int)) Int)
+ **
+ ** (declare-var x Int)
+ ** (declare-var y Int)
+ **
+ ** (constraint (>= (max x y) x))
+ ** (constraint (>= (max x y) y))
+ ** (constraint (or (= x (max x y))
+ **                 (= y (max x y))))
+ ** (constraint (= (+ (max x y) (min x y))
+ **                (+ x y)))
+ **
+ ** (check-synth)
+ **/
```

Comment:
One last request: Can you document what the expected behavior is of the this example?  E.g. this example should print the text: "(define-fun ...)".  Similarly for the other example.

Reply:
Done!